leveldb/table/reader.go

```diff
@@ -615,14 +616,18 @@ func (r *Reader) readBlock(bh blockHandle, verifyChecksum bool) (*block, error)
 	return b, nil
 }
 
-func (r *Reader) readBlockCached(bh blockHandle, verifyChecksum, fillCache bool) (*block, util.Releaser, error) {
-	if r.cache != nil {
+func (r *Reader) readBlockCached(bh blockHandle, verifyChecksum, fillCache bool, metadata bool) (*block, util.Releaser, error) {
+	rcache := r.mcache
+	if !metadata {
```

Comment:
This changes the semantics slightly... If someone wants this to work "as previously", they'd disable metacache. So `r.mcache` is `nil`. Howevever, IIUC, they would _not_ fall back to using `bcache` as "metadata+data cache`, but instead use no cache at all for the metadata. Is that intentional?

Reply:
yes right. I will change it.

Reply:
Done